boxsdk/client/client.py

```diff
@@ -1666,3 +1666,68 @@ def create_metadata_template(self, display_name, fields, template_key=None, hidd
             session=self._session,
             response_object=response,
         )
+
+    @api_call
+    def create_zip(self, name, items):
+        """
+        Creates a zip of multiple files and folders.
+
+        :param name:
+            The name of the zip file to be created.
+        :type name:
+            `unicode`
+        :param items:
+            List of files or folders to be part of the created zip.
+        :type items:
+            `Iterable`
+        :returns:
+            A dictionary representing a created zip
+        :rtype:
+            :class:`dict`
+        """
+        # pylint: disable=protected-access
+        url = self._session.get_url('zip_downloads')
+        data_items = []
+        for item in items:
+            data_items.append({'type': item._item_type, 'id': item.object_id})
+        data = {
+            'download_file_name': name,
+            'items': data_items
+        }
+        response = self._session.post(url, data=json.dumps(data)).json()
+        return self.translator.translate(
+            session=self._session,
+            response_object=response,
+        )
+
+    @api_call
+    def download_zip(self, name, items, writeable_stream):
+        """
+        Creates a zip of multiple files and folders and downloads it.
+
+        :param name:
+            The name of the zip file to be created.
+        :type name:
+            `unicode`
+        :param items:
+            List of files or folders to be part of the created zip.
+        :type items:
+            `Iterable`
+        :param writeable_stream:
+            Stream to pipe the readable stream of the zip file.
+        :type writeable_stream:
+            `zip`
+        :returns:
+            A status response object
+        :rtype:
+            :class:`dict`
+        """
+        created_zip = self.create_zip(name, items)
```

Comment:
This method always calls create_zip.  So, since there's no way for the client to download an already-created zip, then maybe create_zip should be made private?  In that case, maybe rename this method to `download_items_as_zip_file` to be more explicit?

Reply:
Create a zip is a public endpoint that returns information like name conflicts when a zip is created. Some users may want that information, so I think it makes sense to still keep that public.

Reply:
But how would they download the file that was created?

Reply:
They would get the download url from the response of create zip and make a manual api call